File: 1_Two_Sum/C++/solution1.cpp
#include <map>

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::map<int, int> m;
        
        int len = nums.size();
        for(int i = 0; i < len; ++i)
        {
            if(m.find(target - nums[i]) != m.end())
            {
                return vector<int>{i, m[target - nums[i]]};
            }
            m[nums[i]] = i;
        }
        return std::vector<int>{0, 0};
    }
};

File: 1_Two_Sum/Python/solution1.py
class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        attacks = {}
        for i, n in enumerate(nums):
            if target - n in attacks.keys():
                return [i, attacks[target - n]]
            else:
                attacks[n] = i